common/build/eslint-config-fluid/eslint7.js

```diff
@@ -226,6 +227,20 @@ module.exports = {
             }
         ],
 
+        // eslint-plugin-header
+        //
+        // Note: While redundant with the broader header check in 'repo-policy-check', eslint enables early
+        //       detection during 'npm run build' and allows autofixing with both VS Code and 'npm run lint:fix'.
+        "header/header": [
+            "error",
+            "block",
+            {
+                pattern: /^!\n \* Copyright \(c\) Microsoft Corporation\. All rights reserved\.\n \* Licensed under the MIT License\.\n (\*\n \* THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY\n )?$/,
+                template: "!\n * Copyright (c) Microsoft Corporation. All rights reserved.\n * Licensed under the MIT License.\n "
+            },
```

Comment:
If we modify 'gen-version' to place the "THIS IS AN AUTOGENERATED FILE.." warning outside the header comment, we can simplify pattern/template to just a string:
```
        "header/header": [
            "error",
            "block",
            "!\n * Copyright (c) Microsoft Corporation. All rights reserved.\n * Licensed under the MIT License.\n ",
            /* newLines: */ 2, // '\n\n' = 1 blank line after block comment
        ],
```